R/fancy_ideogram.R

```diff
@@ -90,13 +90,13 @@ fancy_ideogram = function(this_sample_id,
   }
 
   #build chr table for segment plotting
-  chr = paste0("chr", c(1:22))
+  chr = c( paste0("chr", c(1:22)), "chrX", "chrY" )
```

Comment:
Yes, the sex chromosomes were purposely left out when setting up many of these fancy plotting functions.

Reply:
I think we can keep it as is, right - just wanted to check because since it is addressed here, it means there was en error regarding that had to be resolved, so wondering to see what was the case.

Reply:
My motivation to check this function was because it was returning many warning messages.
```
> fancy_ideogram(this_sample_id = "DOHH-2",
+                gene_annotation = "MYC",
+                plot_title = "Sample-level Ideogram Example",
+                plot_subtitle = "grch37")
1 region(s) returned for 1 gene(s)
Using the bundled CN segments (.seg) calls in GAMBLR.data...
Using the bundled metadata in GAMBLR.data...
Using the bundled SSM calls (.maf) calls in GAMBLR.data...
Using the bundled metadata in GAMBLR.data...
There were 26 warnings (use warnings() to see them)

> warnings()
Warning messages:
1: In lapply(X = X, FUN = FUN, ...) : NAs introduced by coercion
2: In `[<-.factor`(`*tmp*`, cn_states$CN > 6, value = 6) :
  invalid factor level, NA generated
3: In Ops.factor(cn_states$CN, 6) : ‘>’ not meaningful for factors
4: In fancy_ideogram(this_sample_id = "DOHH-2", gene_annotation = "MYC",  ... :
  NAs introduced by coercion
5: In fancy_ideogram(this_sample_id = "DOHH-2", gene_annotation = "MYC",  ... :
  NAs introduced by coercion
6: In geom_segment(data = segment_data, aes(x = chr_start,  ... :
  Ignoring unknown aesthetics: label
7: In geom_label(data = gene.annotate, aes((x = end - start)/2 +  ... :
  Ignoring unknown parameters: `check_overlap`
8: Width not defined
ℹ Set with `position_dodge(width = ...)`
9: Width not defined
ℹ Set with `position_dodge(width = ...)`
10: Width not defined
ℹ Set with `position_dodge(width = ...)`
11: Width not defined
ℹ Set with `position_dodge(width = ...)`
12: Width not defined
ℹ Set with `position_dodge(width = ...)`
13: Width not defined
ℹ Set with `position_dodge(width = ...)`
14: Width not defined
ℹ Set with `position_dodge(width = ...)`
15: Width not defined
ℹ Set with `position_dodge(width = ...)`
16: Width not defined
ℹ Set with `position_dodge(width = ...)`
17: Width not defined
ℹ Set with `position_dodge(width = ...)`
18: Width not defined
ℹ Set with `position_dodge(width = ...)`
19: Removed 20 rows containing missing values (`geom_segment()`).
20: Removed 20 rows containing missing values (`geom_segment()`).
21: Removed 20 rows containing missing values (`geom_segment()`).
22: Removed 14 rows containing missing values (`geom_segment()`).
23: Removed 14 rows containing missing values (`geom_segment()`).
24: Removed 14 rows containing missing values (`geom_segment()`).
25: Removed 1 rows containing missing values (`geom_text()`).
26: Removed 1 rows containing missing values (`geom_text()`).
```

One of them was because it was trying to convert the strings "X" and "Y" to integers, then getting NAS, in this part:
```
>   #convert data types
>   cols.int = c("start", "end", "ycoord")
>   cn_states[cols.int] = sapply(cn_states[cols.int], as.integer)
Warning message:
In lapply(X = X, FUN = FUN, ...) : NAs introduced by coercion
```

I fix all warning messages and now the function works 100% clean while creating the same plot.

Then I wanted to add the sexual chromosomes and run an example that shows variants in the X chromosome but not in the Y. 

My idea was to make the function check and remove any chromosome with no variants --- this implementation I haven't done yet. Do you think this could be good, or just always removing the Y chrom would be enough?

Reply:
The X chromosome was added and the Y was ignored.